worlds/smw/Regions.py

```diff
@@ -2162,23 +2161,10 @@ def add_location_to_region(multiworld: MultiWorld, player: int, active_locations
             add_rule(location, rule)
 
 
-
-def connect(multiworld: MultiWorld, player: int, used_names: typing.Dict[str, int], source: str, target: str,
+def connect(world: World, source: str, target: str,
             rule: typing.Optional[typing.Callable] = None):
-    source_region = multiworld.get_region(source, player)
-    target_region = multiworld.get_region(target, player)
-
-    if target not in used_names:
-        used_names[target] = 1
-        name = target
-    else:
-        used_names[target] += 1
-        name = target + (' ' * used_names[target])
-
-    connection = Entrance(player, name, source_region)
-
+    source_region: Region = world.multiworld.get_region(source, world.player)
+    connection_rules: typing.Dict[str, typing.Callable[[CollectionState], bool]] = dict()
     if rule:
-        connection.access_rule = rule
-
-    source_region.exits.append(connection)
-    connection.connect(target_region)
+        connection_rules[target] = rule
+    source_region.add_exits([target], connection_rules)
```

Comment:
this should really be using `Region.connect` instead of `Region.add_exits`
```suggestion
    target_region: Region = world.multiworld.get_region(target, world.player)
    source_region.connect(target_region, rule)
```
 there are also getter helpers on the World now so you can just do `target_region: Region = world.get_region(target)`